pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -111,6 +112,7 @@ private static void setOptions(PinotQuery pinotQuery, List<String> optionsStatem
   private static PinotQuery compileCalciteSqlToPinotQuery(String sql) {
     SqlParser.ConfigBuilder parserBuilder = SqlParser.configBuilder();
     parserBuilder.setLex(PINOT_LEX);
+    parserBuilder.setConformance(SqlConformanceEnum.LENIENT);
```

Comment:
Why is this needed?

Reply:
I think I committed another change for supporting `LIMIT offset,numRecords` into this.

Reply:
basically calcite parser cannot parse `LIMIT X,Y` until this conformance is added.

Reply:
> This is supported on all paths right?
> 
> (1) PQL -> PqlCompiler -> BrokerRequest
> (2) PQL -> PqlCompiler -> BrokerRequest -> PinotQuery -> brokerRequest.setPinotQuery()
> (3) SQL -> Calcite parser -> PinotQuery

For path 3 it's SQL -> Calcite parser -> PinotQuery -> BrokerRequest
Query processing still depends on BrokerRequest.

Also plan to make query console default to go through path 3 in PR https://github.com/apache/incubator-pinot/pull/4994

Reply:
@siddharthteotia we did an exercise at LI wherein we passed all possible used queries though Calcite parser. We caught certain things which do not pass under the default sql conformance level (`!=` and `LIMIT offset, count`). One of the solutions for that we discussed was to set the conformance to be more lenient.

Reply:
Got it. Thanks